README.md

```diff
@@ -2,20 +2,175 @@
 
 ```
-> npm ci
-> npm test
+src
+├─ user
+|     ├─ __snapshots__
+|     |     └─ user.spec.js.snap 
+|     └─ user.spec.js
+├─ job
+¦     └─ job.spec.js
+¦
+¦
+├─ scenarios
+|     ├─ __snapshots__
+¦     ¦     ¦
+¦     ¦     ├─ revertVm.spec.js.snap 
+¦     ¦     └─ migrateVm.spec.js.snap
+|     ├─ revertVm.spec.js
+|     └─ migrateVm.spec.js
+├─ _xoWithTestHelpers.js
+└─ util.js
 ```
 
+There are two ways to classify tests:
+
+  + By XO objects.  
+    In each file `[XO object].spec.js`, the tests must be grouped by methods.  
+    
+    Let's take an example with the XO object **user** and some of its methods:  
+    ```
+    describe("user", () => {
+      describe(".create() :", () => {
+        it("creates a user with permission", async () => {
+          /* some tests */
+        });
+        it("creates a user without permission", async () => {
+          /* some tests */
+        });
+      });
+
+      describe(".delete() :", () => {
+        it("deletes a user successfully with id", async () => {
+          /* some tests */
+        });
+      });
+    });
+    ```
+    The test must be simple : one feature by test.  
+
+  + By scenarios.  
+    In the folder `scenarios`: name the test files with what they do *mainly*.  
+    Be as precise as possible in the title of the test.  
+    
+    For example, in the `revertVm.spec.js` file:  
+    ```
+    describe("revertVm :", {
+      it("fails trying to execute three times a rolling snapshot with 2 as retention and reverts the VM to the first snapshot which is normally removed", async () => {
+        /* some tests */
+      });
+    });
+    ```
+    
+### Test environment
+
+- The test environment must remain the same before and after each test:  
+  * each resource created must be deleted  
+  * no existing resource before running the tests should be deleted or modified.  
+  
+  There is a helper `_xoWithTestHelpers.js` to manage these temporary resources.  
+
+### Implementation
+
+- Use `_xoWithTestHelpers.js` to create or delete resources.
+ 
+- Import `util.js` to use the xo connection and all the methods that could be used on xo.
+
+- To generate TempId, use `_randomId.js`.
+
+- Take values ​​that cover the maximum of testing possibilities.
+
+- Make a sentence for the title of the test. It must be clear and consistent. For example:  
+  `it("creates a backupNg job without schedule", () => { /* some tests */ });`  
+  `it("fails trying to run a backupNg job without schedule", () => { /* some tests */ });`  
+
+- Use the javascript testing framework **jest**.  
+  On the first line of each test file, set the environment: `/* eslint-env jest */`.  
+
+- If the feature you want to test is not implemented : write it and skip it, using `it.skip()`.
+
+- If you want to do something before/after each test or all tests put it in `util.js` by using `afterEach()`, `beforeAll()`...  
```

Comment:
That depends on the cases. it can also be defined locally to a XO object tests. It's a developer decision not a guideline.

Reply:
That's why I  specify before ALL tests , but I'll remove it :+1:  .